mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/utils/GridFsUtil.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.mongodb.utils;
+
+import java.io.ByteArrayOutputStream;
+import java.lang.invoke.MethodHandles;
+import java.lang.reflect.Field;
+
+import org.hibernate.engine.jdbc.BinaryStream;
+import org.hibernate.ogm.datastore.mongodb.logging.impl.Log;
+import org.hibernate.ogm.datastore.mongodb.logging.impl.LoggerFactory;
+import org.hibernate.ogm.datastore.mongodb.options.BinaryStorageType;
+import org.hibernate.ogm.datastore.mongodb.options.impl.BinaryStorageOption;
+import org.hibernate.ogm.datastore.mongodb.options.impl.GridFSBucketOption;
+import org.hibernate.ogm.model.key.spi.EntityKey;
+import org.hibernate.ogm.model.spi.Tuple;
+import org.hibernate.ogm.model.spi.Tuple.SnapshotType;
+import org.hibernate.ogm.options.spi.OptionsContext;
+import org.hibernate.ogm.options.spi.OptionsService;
+
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoCursor;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.gridfs.GridFSBucket;
+import com.mongodb.client.gridfs.GridFSBuckets;
+import org.bson.Document;
+import org.bson.types.ObjectId;
+
+/**
+ * @author Sergey Chernolyas &amp;sergey_chernolyas@gmail.com&amp;
+ */
+public class GridFsUtil {
+	private static final Log log = LoggerFactory.make( MethodHandles.lookup() );
+
+	public static void storeContentToGridFs(MongoDatabase mongoDatabase, Document currentDocument,
+											EntityKey entityKey, OptionsService optionService, Tuple tuple, SnapshotType snapshotType) {
+		Class entityClass = TableEntityTypeMappingInfo.getEntityClass( entityKey.getTable() );
+		if ( currentDocument == null ) {
+			return;
+		}
+		for ( String fieldName : currentDocument.keySet() ) {
+			if ( fieldName.equals( "$set" ) ) {
+				// it is part of request. it is not document
+				Document queryFields = (Document) currentDocument.get( fieldName );
+				for ( String queryField : queryFields.keySet() ) {
+					loadContent( mongoDatabase,queryFields,entityClass,queryField,optionService,tuple,snapshotType );
+				}
+			}
+			else {
+				// it is not document
+				loadContent( mongoDatabase,currentDocument,entityClass,fieldName,optionService,tuple,snapshotType );
+			}
+		}
+	}
+
+	private static void loadContent(MongoDatabase mongoDatabase,  Document currentDocument, Class entityClass, String fieldName,
```

Comment:
This method is called `loadContent` but it does all sort of things

Reply:
What do you mean?

Reply:
I find this code hard to follow. 

The method is called `loadContent`, that it is already quite generic, there is no javadoc  and when I try to figure out what's going on I see that it also updates some document and delete stuff from the bucket.
This class needs a JavaDoc explaining what's going on and explaining briefly what's the purpose of GridFS.

The fact that each method starts with
```
if ( BinaryStorageType.GRID_FS == binaryStorageType ) {
```

makes me think that there is a layer of abstraction missing.
I think we need an additional class, let's call it `BinaryStorageManager`, that decides which `BinaryStorageType` we need and delegates the execution to a `BinaryStorageDelegator` interface; in this case it would be a `GridFSDelegator`.
It makes sense if we use a `BinaryStorageType` enum and,  we can also avoid using static methods in favor of  a proper object oriented approach.